### DISCUSSION TITLE
Issue with COG and INIVEL w and w/o LS-DYNA input

### DISCUSSION THREAD

**Discussion by SonSiont** [OpenRadioss/OpenRadioss#1790] [Q&A]
**Describe the bug**
The LS-DYNA model contains rigid bodies with initial velocities.
The COG values that run with the LS-DYNA input are bad.
The COGs will be calculated correctly if you set the coordinate to (0,0,0) in my own translation "conrad_0000.rad".
In both cases, the translation velocities are wrong.

**To Reproduce**
[cog.tar.gz](https://github.com/OpenRadioss/OpenRadioss/files/11358695/cog.tar.gz)

Run both input files, cog6.k and conrad_0000.rad .

Compare the coordinates and velicites of node 603948 and 603949

**Expected behavior**
Correct velocities.
If ICoG is set to "1" in /RBODY, the initial velocities should be applied after calculating the COGs.

**Output**
Screenshots
Marked potential COG:
![cog3](https://user-images.githubusercontent.com/80820850/235292744-069b9801-379f-4b1f-966b-4d8d04596f0c.png)
comparison of coordinates:
![cog2](https://user-images.githubusercontent.com/80820850/235292763-ef4dcd8b-7fc1-41b2-8955-fa04c434faa8.png)
comparison of velocities:
![cog](https://user-images.githubusercontent.com/80820850/235292784-ff2eb972-cdd4-4782-b6c8-220144d8d8b3.png)
**Desktop (please complete the following information):**
- Version [f1e5167efe7d9e7b190ca73aa33f70307234fe15]
- Hardware [Linux; Intel(R) Core(TM) i7-5930K CPU @ 3.50GHz (x86_64), 3499 MHz,  64217 MB RAM]
- Configuration [Double precision SMP, one task]
**Additional context**
In the model conrad_0000.rad the node 603948 is set to the correct position.

**Comment by SonSiont** [comment]
Hi,
the situation gets worse if you use the options *CONSTRAINED_RIGID_BODIES / /MERGE/RBODY.

**Comment by PaulAltair** [comment]
I've taken a look at your model, and the reporting is a little odd here, and we can take a look at it, but I'm not sure this really 'matters' as such for the solution?, it is just an artefact of the generated location of the nodes, but since they are moved to the solver calculated CofG anyway they are in the 'correct' location once the engine starts? the nodes in question don't exist in the Dyna model, and are arbitrarily created for the RBODY main nodes in Radioss, Once the model is running (cog6) the XYZ velocities for 603948 are 6915.536, 0.000 and -237.252 for 603949 they are  7016.214, 0.0, -245.365 so different to what the starter reports (and your modified version)

The nodes of the discs themselves match almost perfectly, If I run cog6 up to 0.1s for the model (forcing max ts of 1e-6 in both Dyna and Radioss) then the nodal velocities of the discs are identical throughout (to 2 or 3 decimal places)

**Comment by SonSiont** [comment]
Hi Paul,

I found differences in the rotational energy in a larger model and I cooked it down.
![Bildschirmfoto_2023-05-08_13-13-31](https://user-images.githubusercontent.com/80820850/236810549-5934757a-ffe6-4231-b3bb-2283ae43711f.png)
The rotational energy was 8055 and with the changes in the cog it jumps to 8184.
best regards

**Comment by elequiniou** [comment]
Moving to discussion, as no clear issue identified, more a precision issue on rotational calculation.
